test/integration/singlecluster/scheduler/fairsharing/fair_sharing_test.go

```diff
@@ -295,8 +296,140 @@ var _ = ginkgo.Describe("Scheduler", func() {
+			cqLeft            *kueue.ClusterQueue
+			cqRight           *kueue.ClusterQueue
+			lqLeft            *kueue.LocalQueue
+			lqRight           *kueue.LocalQueue
+		)
+
+		ginkgo.BeforeEach(func() {
+			//         root
+			//        /    \
+			//       fl     fr
+			//     /   \     \
+			//    sl   sr    bank
+			//    /     \
+			//   cqL    cqR
+
+			cohortFirstLeft = testing.MakeCohort("first-left").Parent("root").Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohortFirstLeft)).To(gomega.Succeed())
+
+			cohortFirstRight = testing.MakeCohort("first-right").Parent("root").Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohortFirstRight)).To(gomega.Succeed())
+
+			cohortSecondLeft = testing.MakeCohort("second-left").Parent(kueue.CohortReference(cohortFirstLeft.Name)).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohortSecondLeft)).To(gomega.Succeed())
+
+			cohortSecondRight = testing.MakeCohort("second-right").Parent(kueue.CohortReference(cohortFirstLeft.Name)).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohortSecondRight)).To(gomega.Succeed())
+
+			cohortBank = testing.MakeCohort("bank").Parent(kueue.CohortReference(cohortFirstRight.Name)).
+				ResourceGroup(
+					testing.MakeFlavorQuotas(defaultFlavor.Name).Resource(corev1.ResourceCPU, "10").FlavorQuotas,
+				).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohortBank)).To(gomega.Succeed())
+
+			cqLeft = testing.MakeClusterQueue("left").
+				Cohort(kueue.CohortReference(cohortSecondLeft.Name)).
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(defaultFlavor.Name).Resource(corev1.ResourceCPU, "2").Obj(),
+				).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqLeft)).To(gomega.Succeed())
+
+			cqRight = testing.MakeClusterQueue("right").
+				Cohort(kueue.CohortReference(cohortSecondRight.Name)).
+				ResourceGroup(
+					*testing.MakeFlavorQuotas(defaultFlavor.Name).Resource(corev1.ResourceCPU, "2").Obj(),
+				).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cqRight)).To(gomega.Succeed())
+
+			lqLeft = testing.MakeLocalQueue("left", ns.Name).ClusterQueue(cqLeft.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, lqLeft)).To(gomega.Succeed())
+
+			lqRight = testing.MakeLocalQueue("right", ns.Name).ClusterQueue(cqRight.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, lqRight)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cqLeft, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cqRight, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cohortFirstLeft, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cohortFirstRight, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cohortSecondLeft, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cohortSecondRight, true)
+			util.ExpectObjectToBeDeleted(ctx, k8sClient, cohortBank, true)
+		})
+
+		ginkgo.It("Admits workloads respecting fair share", func() {
```

Comment:
how about adding workloads to each CQ in a round robin fashion? Then, we should see even fair sharing scores, at each level of the hierarchy

Reply:
Done